src/services/jsTyping.ts

```diff
@@ -46,8 +55,44 @@ namespace ts.JsTyping {
             }
         }
         
-        let newTypingNames = typingNames.filter(t => cachedTypingNames.indexOf(t) < 0);
-        
+        let newTypingNames = ts.filter(typingNames, t => cachedTypingNames.indexOf(t) < 0);
         return { cachedTypingPaths, newTypings: newTypingNames };
     }
+
+    export function updateTypingsConfig(
+        host: TypeDefinitionResolutionHost | System, cachedTypingsPaths: string[], newTypings: string[], cachePath: string, typingsConfigPath: string): void {
+        let tsdJsonCacheInfo = tryGetTsdJsonCacheInfo(host, cachePath);
+        if (tsdJsonCacheInfo) {
+            let notFound: string[] = [];
+            let cacheTsdJsonDict = tsdJsonCacheInfo.value;
+            if (cacheTsdJsonDict.hasOwnProperty("installed")) {
+                let installedTypings = Object.keys(cacheTsdJsonDict.installed);
+                notFound = ts.filter(newTypings, i => !isInstalled(i, installedTypings));
+            }
+            if (cacheTsdJsonDict.hasOwnProperty("notFound")) {
+                notFound = ts.deduplicate<string>(cacheTsdJsonDict["notFound"].concat(notFound));
+            }
+            if (notFound.length > 0) {
+                cacheTsdJsonDict["notFound"] = notFound;
+                host.writeFile(tsdJsonCacheInfo.path, JSON.stringify(cacheTsdJsonDict));

```

Comment:
What if there was originally some not found items in the json file -> then the repo gets updated -> then the new notFound becomes 0 -> this is not called, and the json file is outdated. 

I think we should reset the notFound for every period of time